descriptor.go

```diff
@@ -1437,3 +1445,733 @@ func parseDescriptors(i *astikit.BytesIterator) (o []*Descriptor, err error) {
 	}
 	return
 }
+
+// TODO (barbashov) move it to BitsWriter?
+func writeBytesN(b *astikit.BitsWriterBatch, bs []byte, n int, padByte uint8) {
+	if len(bs) >= n {
+		b.Write(bs[:n])
+	} else {
+		b.Write(bs)
+		b.Write(bytes.Repeat([]byte{padByte}, n-len(bs)))
+	}
+}
+
```

Comment:
I would make all below functions part of descriptor specific structs.

For instance `func calcDescriptorAC3Length(d *DescriptorAC3) uint8` would become `func (d *DescriptorAC3) calcLength() uint8` and `func writeDescriptorAC3(w *astikit.BitsWriter, d *DescriptorAC3) error` would become `func (d *DescriptorAC3)write(w *astikit.BitsWriter) error`

Reply:
I though about that.

If we do this here, I suppose we'll have to do it with all other functions in muxer and demuxer as well - for consistency. That's quite a lot of refactoring.

So I decided against it for the time being.

Reply:
ok fair enough, let's save this for later, this PR is big enough